.github/workflows/python-app.yml

```diff
@@ -23,10 +23,18 @@ jobs:
       uses: actions/setup-python@v3
       with:
         python-version: "3.10"
+    - name: Cache dependencies
+      uses: actions/cache@v2
```

Comment:
Is there any particular reason to use v2 instead of v3 here and below?

Reply:
Actually, the difference isn't breaking at all based on the readme of the action/cache repo. PTLA https://github.com/actions/cache
Do you want me to go ahead and change it?

Reply:
Yes, please, let's use the latest one.

Reply:
done